mysql/position.go

```diff
@@ -6,11 +6,15 @@ import (
 
 // For binlog filename + position based replication
 type Position struct {
```

Comment:
I think `Position` is just binlog position,  if you need , I think it' better to create a new sturcture that contains server id and position

Reply:
But application that implements `canal.EventHandler` depends on `mysql.Position`, and some application need to know current MySQL master's server id,  to have a unique map between application's data:
```
 (server_id, name, pos) ---> data
``` 
Only this way , application can distinguish it's data when MySQL failover/switchover.

Reply:
I append a commit 503c6b4 , add the logic about telling the current server_id to event handler (applications).

Reply:
what do you think about it @siddontang ?

Reply:
I append a commit 12d3dda, fix comparison between different server_id in  `c.WaitUntilPos()` . I think a ha application need to care about MySQL failover/switchover.

Reply:
IMO, it is confused to add a ServerID in the Position structure, maybe we should use another way like checking server ID explicitly and separately.

Reply:
Well, i drop the logic about server id. Now there are only logic about copy of GTIDSet. I call `Clone()` in `Canal.master.GTIDSet()` and `BinlogSyncer.getGtidSet()` to return copy of GTIDSet. PTAL.